test/integration/test_utils.js

```diff
@@ -97,16 +97,31 @@ function getSelectedEditors(page) {
 }
 exports.getSelectedEditors = getSelectedEditors;
 
-async function waitForEvent(page, eventName, timeout = 30000) {
-  await Promise.race([
+async function waitForEvent(page, eventName, timeout = 1000) {
+  const hasTimedout = await Promise.race([
     // add event listener and wait for event to fire before returning
-    page.evaluate(name => {
-      return new Promise(resolve => {
-        document.addEventListener(name, resolve, { once: true });
-      });
-    }, eventName),
-    page.waitForTimeout(timeout),
+    page.evaluate(
+      name =>
+        new Promise(resolve => {
+          document.addEventListener(name, () => resolve(false), { once: true });
+        }),
+      eventName
+    ),
+    page.evaluate(
+      (timeOut, name) =>
+        new Promise(resolve => {
+          setTimeout(() => {
+            console.log(`Timeout waiting for ${name}`);
```

Comment:
Given the logging below, why do we need this line as well?
If there's a good reason, please add a (short) comment explaining why.